hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/metadata/SigningCertFromMetadataExtractor.java

```diff
@@ -48,7 +48,10 @@ public X509Certificate getSigningCertForCurrentSigningKey(PublicKey publicSignin
         try {
             Credential credential = credentialResolver.resolveSingle(criteriaSet);
             if (credential instanceof X509Credential) {
-                return ((X509Credential)credential).getEntityCertificate();
+                X509Certificate x509Cert = ((X509Credential)credential).getEntityCertificate();
+                if (x509Cert.getPublicKey().equals(publicSigningKey)){
```

Comment:
minor style thing that you can feel free to ignore - might be better to swap the check around as publicSigningKey will never be null